docs/src/md/kotlin.core/declarations.md

```diff
@@ -306,86 +306,6 @@ Inner classes cannot be declared in [object declarations][Object declaration], a
 > }
 > ```
 
-##### Inheritance delegation
```

Comment:
You made a great point in the issue, that this section is somewhere in between `Inheritance` and `Declarations`. One of the ways we use in the specification to reconcile such cases is to keep a forwards or backwards reference, saying "Well, actually we also have inheritance delegation, it is described in more detail [here][there]". So that the reader who comes to `Declarations` looking for inheritance delegation still can find the full description.

Reply:
The main question to answer here is then: where should we have the actual section and where should we have the reference? In `Declarations` or in `Inheritance`? I would like you to think a bit more, make a decision and explain your reasoning.

Reply:
On the one hand, it seems that the chapter `Inheritance` talks mainly about inheritance and redefinition rules, rather than specific cases, while `Inheritance Delegation` is a specific thing and syntax related to classifier declaration. 
However, on the other hand, the previous location of `Inheritance Delegation` is hidden deep inside the class declaration and includes both the syntax for it (which is suitable for this chapter) and the inheritance logic with rules (which is more suitable for the chapter `Inheritance`). In addition, it uses some of the concepts presented later in the chapter `Inheritance`. 

One simple solution in this case would be to somehow split the entire `Inheritance Delegation` section into two parts-one with syntax, an example (perhaps some technical details) in the `Declarations` chapter, where it is present right now, and another referring to the exact chapters and concepts from the section `Inheritance` could go to the `Inheritance` section itself. These two "parts" could be further connected to each other.

However, the most significant problem with this approach is that the section itself is quite small, and splitting it will result in two even smaller sections with incomplete information, and in order to understand one of them, the reader will need to read the other anyway.

If the "Split" approach is acceptable in this situation, then I would prefer it (even though how to do the splitting is a good question), as the section does not fully belong to either of the chapters.

Reply:
Thanks! I also agree that the split would be quite difficult. I think that we should keep the body of the section in "Declarations" and simply create the same-named section with reference to it in "Inheritance". WDYT?

Reply:
Sounds good to me, will do so. 
How much should be put to the section in inheritance? Just a reference will be too little for a section, and, I think, even for a subsection. If duplicating information is fine, then I guess I can put two first paragraphs of the original subsection into the new one, with third being somewhere around "Technical details and examples can be found in <REF>". Or even put only the first one along with a small syntax example and just a few sentences from the second one, along with the reference...

Reply:
It can be very minimal, for example, we have this very short section: https://kotlinlang.org/spec/built-in-types-and-their-semantics.html#built-in-annotation-types =)

But feel free to propose a more suitable split or reference.